src/contexts/EthereumContext.tsx

```diff
@@ -268,19 +265,30 @@ export function EthereumProvider({ children }: { children: React.ReactNode }) {
     }
     // Regular Connection
     else {
-      if (!provider) {
+      /*
+       * Check if the Universal Profile extension or regular
+       * wallet injected the related window object
+       */
+      const providerObject = window.lukso || window.ethereum;
+
+      // Set global provider
+      if (providerObject) {
+        const plainProvider = new ethers.BrowserProvider(providerObject);
+
+        try {
+          const accounts = await plainProvider.send('eth_requestAccounts', []);
```

Comment:
Don't you want to try eth_getAccounts() first in case the user previously connected to this dapp?

Reply:
Fantastic idea! I definitely should do that. Also created a `connectAccount()` function to not repeat code for all 3 connection setups. Thanks 🙏🏻